osquery/tables/system/windows/system_info.cpp

```diff
@@ -45,17 +46,40 @@ QueryData genSystemInfo(QueryContext& context) {
     r["cpu_logical_cores"] = INTEGER(numProcs);
     wmiResultsProc[0].GetLong("NumberOfCores", numProcs);
     r["cpu_physical_cores"] = INTEGER(numProcs);
-    wmiResults[0].GetString("TotalPhysicalMemory", r["physical_memory"]);
     wmiResults[0].GetString("Manufacturer", r["hardware_vendor"]);
     wmiResults[0].GetString("Model", r["hardware_model"]);
   } else {
     r["cpu_logical_cores"] = "-1";
     r["cpu_physical_cores"] = "-1";
-    r["physical_memory"] = "-1";
     r["hardware_vendor"] = "-1";
     r["hardware_model"] = "-1";
   }
 
+  // Physical memory size is reported from Win32_PhysicalMemory. See
+  // TotalPhysicalMemory at
+  // https://docs.microsoft.com/en-us/windows/win32/cimwin32prov/win32-computersystem
+  const WmiRequest wmiSystemReqMem("select Capacity from Win32_PhysicalMemory");
+  const std::vector<WmiResultItem>& wmiResultsMem = wmiSystemReqMem.results();
+  if (wmiResultsMem.empty()) {
+    r["physical_memory"] = "-1";
+  } else {
+    // capacity is documented as being a uint64, but it actually coming back as
+    // a string.
+    unsigned __int64 sum = 0;
+    std::string capacityStr = "0";
+    for (const auto& wmiResult : wmiResultsMem) {
+      auto s = wmiResult.GetString("Capacity", capacityStr);
+      if (s.ok() == false) {
+        VLOG(1) << "Warning unable to get memory module capacity value from "
+                   "wmi results: "
+                << s.getMessage();
+        continue;
```

Comment:
My guess is this case does not happen often and we can choose to set the value to `-1` indicating we don't know what is going on? Thus `break`ing from the loop.

Reply:
No idea. I wasn't sure what we'd see on a machine with 4 slots with only 2 populated. (My windows test machines are laptops)

Reply:
I mostly left this as is, since it's an unknown to me.

Reply:
Is a WMI query necessary here or would a standard Windows API function call suffice? The function `GlobalMemoryStatusEx` populates a `MEMORYSTATUSEX` structure containing the `ullTotalPhys` field.

On a side note, I am back from hiatus. :)

Link: [GlobalMemoryStatusEx](https://docs.microsoft.com/en-us/windows/win32/api/sysinfoapi/nf-sysinfoapi-globalmemorystatusex)

Reply:
Nice to see you back!

I made a quick implementation for `GlobalMemoryStatusEx`. It did not return the actual total physical ram. Instead, it returns the same value as `Win32_ComputerSystem -> TotalPhysicalMemory`, which I believe is the ram not held by the bios.

On this 8 gig machine:
* GlobalMemoryStatusEx: 7896399872
* wmi TotalPhysicalMemory: 7896399872
* wmi sum of the capacity (as this PR stands): 8589934592

I guess this raises a bit of a philosophical question about what "total physical ram" is. But ultimately, I think windows is wrong here. We document physical ram, so we should report it.

Reply:
Aha. You want `GetPhysicallyInstalledSystemMemory`: https://docs.microsoft.com/en-us/windows/win32/api/sysinfoapi/nf-sysinfoapi-getphysicallyinstalledsystemmemory

Reply:
Oh, sweet. I've updated this.

I suspect all the data for this table is better sourced via that API and not WMI. But that's a bigger change